conformance/tests/admin-network-policy-core-egress-tcp-rules.go

```diff
@@ -207,12 +208,65 @@ var AdminNetworkPolicyEgressTCP = suite.ConformanceTest{
 			// ensure egress from gryffindor is PASSED to slytherin at port 80; egressRule at index3 should take effect
 			success := kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-0", "tcp",
 				serverPod.Status.PodIP, int32(80), s.TimeoutConfig.RequestTimeout, true)
-			assert.Equal(t, true, success)
+			assert.True(t, success)
 			// harry-potter-1 is our client pod in gryffindor namespace
 			// ensure egress from gryffindor is ALLOWED to slytherin for rest of the traffic; matches no rules hence allowed
 			success = kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-gryffindor", "harry-potter-1", "tcp",
 				serverPod.Status.PodIP, int32(8080), s.TimeoutConfig.RequestTimeout, true)
-			assert.Equal(t, true, success)
+			assert.True(t, success)
 		})
 	},
 }
+
+var AdminNetworkPolicyEgressNamedPort = suite.ConformanceTest{
+	ShortName:   "AdminNetworkPolicyEgressNamedPort",
+	Description: "Tests support for egress traffic on a named port using admin network policy API based on a server and client model",
+	Features: []suite.SupportedFeature{
+		suite.SupportAdminNetworkPolicy,
+		suite.SupportAdminNetworkPolicyNamedPorts,
+	},
+	Manifests: []string{"base/admin_network_policy/core-egress-tcp-rules.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+
+		t.Run("Should support an 'allow-egress' policy for named port", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), s.TimeoutConfig.GetTimeout)
+			defer cancel()
+			// This test uses `egress-tcp` ANP
+			// cedric-diggory-1 is our server pod in hufflepuff namespace
+			serverPod := &v1.Pod{}
+			err := s.Client.Get(ctx, client.ObjectKey{
+				Namespace: "network-policy-conformance-hufflepuff",
+				Name:      "cedric-diggory-1",
+			}, serverPod)
+			require.NoErrorf(t, err, "unable to fetch the server pod")
+			anp := &v1alpha1.AdminNetworkPolicy{}
+			err = s.Client.Get(ctx, client.ObjectKey{
+				Name: "egress-tcp",
+			}, anp)
+			require.NoErrorf(t, err, "unable to fetch the admin network policy")
+			namedPortRule := anp.DeepCopy().Spec.Egress[5]
+			webPort := "web"
+			// replace the tcp port 8080 rule as named port rule which translate to tcp port 80 instead
+			namedPortRule.Ports = &[]v1alpha1.AdminNetworkPolicyPort{
+				{
+					NamedPort: &webPort,
```

Comment:
I am ok with doing this but just wondering if adding a rule to the existing yaml with the namedPort is better? -> It could serve as an example of how a namedPort ANP should be created for the users? I won't hold the PR for that thinking out loud on what is better moving forward.

Reply:
While I agree with this, I think if we decided to go down that path we should actually have a separate yaml spec for each of the policy that we use and maybe avoid hot patching the policies in the tests altogether :) in terms of named port, I don't feel it is a feature that we need special documentation/example out of.